### PR TITLE
Adding API to add and delete Nuage ports to and from alubr0 bridge

### PR DESCRIPTION
**Pull request by rparulek**
Adding API to add and delete Nuage ports to and from alubr0 bridge

### CODE REVIEW

File: api/Port.go [hunk 1]
```diff
@@ -276,3 +285,118 @@ func (vrsConnection VRSConnection) processUpdates(updates *libovsdb.TableUpdates
 	}
 	return nil
 }
+
+// AddPortToAlubr0 adds Nuage port to alubr0 bridge
+func AddPortToAlubr0(ovs *libovsdb.OvsdbClient, intfName string, entityName string, entityUUID string) error {
```
Comment (vareti) [inline api/Port.go]: 1) OvsdbClient structure is [private](https://github.com/nuagenetworks/libvrsdk/blob/master/api/VRSConnection.go#L23) to "api" package. plugins code will be not be able to see it. I am thinking this function should be defined on this [structure](https://github.com/nuagenetworks/libvrsdk/blob/master/api/VRSConnection.go#L22) just like other functions.

2) I think it is better to pass function arguments as a structure. Helps us in extending the functionality without breaking the API.

File: api/Port.go [hunk 2]
```diff
@@ -276,3 +285,118 @@ func (vrsConnection VRSConnection) processUpdates(updates *libovsdb.TableUpdates
...
+	reply, err := ovs.Transact(OvsDBName, operations...)
+	if err != nil || len(reply) < len(operations) {
+		return fmt.Errorf("Problem mutating row in the OVSDB Bridge table for alubr0")
+	}
+
+	return nil
+}
+
+// RemovePortFromAlubr0 will remove a port from alubr0 bridge
+func RemovePortFromAlubr0(ovs *libovsdb.OvsdbClient, portName string) error {
```
Comment (vareti) [inline api/Port.go]: Same comments that I added previously

1) OvsdbClient structure is [private](https://github.com/nuagenetworks/libvrsdk/blob/master/api/VRSConnection.go#L23) to "api" package. plugins code will be not be able to see it. I am thinking this function should be defined on this [structure](https://github.com/nuagenetworks/libvrsdk/blob/master/api/VRSConnection.go#L22) just like other functions.

2) I think it is better to pass function arguments as a structure. Helps us in extending the functionality without breaking the API.

Comment (rparulek) [reply api/Port.go]: Indeed. I was working on making the same change while you added the comment. Thanks for pointing this out.

File: api/Port.go (hunk 1 above)
Comment (rparulek) [reply api/Port.go]: Indeed. I was working on making the same change while you added the comment. Thanks for pointing this out.

File: api/Port.go [hunk 3]
```diff
@@ -276,3 +286,118 @@ func (vrsConnection VRSConnection) processUpdates(updates *libovsdb.TableUpdates
 	}
 	return nil
 }
+
+// AddPortToAlubr0 adds Nuage port to alubr0 bridge
+func (vrsConnection *VRSConnection) AddPortToAlubr0(intfName string, entityName string, entityUUID string) error {
```
Comment (vareti) [inline api/Port.go]: thinking out loud..  i am okay with passing the intfName directly as string but the other two i am thinking need to be passed as a struct. At a later point if we want to add one more argument to the function, we can pass it as part of structure. That will not the break the code of the users who are already using the API.

File: api/Port.go [hunk 4]
```diff
@@ -276,3 +286,118 @@ func (vrsConnection VRSConnection) processUpdates(updates *libovsdb.TableUpdates
...
+
+	// simple mutate operation
+	mutateOp := libovsdb.Operation{
+		Op:        "mutate",
+		Table:     "Bridge",
+		Mutations: []interface{}{mutation},
+		Where:     []interface{}{condition},
+	}
+
+	operations := []libovsdb.Operation{deleteOp, mutateOp}
```
Comment (vareti) [inline api/Port.go:396]: quick question.. i see that only port and bridge tables are cleaned up. what about interface tables?

Comment (rparulek) [reply api/Port.go:396]: @vareti We do not need to explicitly make an OVSDB operation to remove entry from the interface table. When a port entry is deleted from the port table, OVS removes the interfaces that port is referencing from the Interface table automagically. This is the reason we only cleanup from Port and Bridge table. I have tested this behaviour using our SDK tests and see the interface table entries getting cleaned up after entry gets removed from Port table.

**Review by vareti — APPROVED** [review]
looks good to me

**Pull request merged**